src/game/boe.specials.cpp

```diff
@@ -4150,7 +4156,7 @@ void townmode_spec(const runtime_state& ctx) {
 				showError("Invalid attitude (0-Friendly Docile, 1-Hostile A, 2-Friendly Will Fight, 3-Hostile B).");
 				break;
 			}
-			univ.town.monst[spec.ex1a].attitude = eAttitude(spec.ex1b);
+			univ.town.monst.at(spec.ex1a).attitude = eAttitude(spec.ex1b);
```

Comment:
Using `at` here doesn't help anything. It still crashes, just reliably (ie, every time) and sooner (ie, not later on because of data corruption). What's really needed is a manual range check… which means you might as well stick to the `[]` operator.